### PR TITLE
feat(vendor-finance): Adds support for v2 ETH pools + swaps subgraph endpoints

### PR DESCRIPTION
**Pull request by ForrestChew**
## Description
This PR contains two non-insignificant changes.
1. Adds support for v2 pools on Ethereum
2. Swaps subgraph URLs in favor of custom entity fetcher URL for all networks.

## Checklist

- [x] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] (optional) As a contributor, my Ethereum address/ENS is:
- [x] (optional) As a contributor, my Twitter handle is: @BossMcBara

## How to test?
To see that the new entity fetcher URL works follow the below:
1. Run `pnpm dev`
2. Verify V2 on Ethereum: `http://localhost:5001/apps/vendor-finance/balances?addresses%5B%5D=0xC6c48E582405BE32f25EdBa211CCaa820058Cb2f&network=ethereum`
3. Verify V2 on Arbitrum: `http://localhost:5001/apps/vendor-finance/balances?addresses%5B%5D=0x2a6334c30f36d1eefd64689c590dd199d3475972&network=arbitrum`
4. Verify v1 on Ethereum: `http://localhost:5001/apps/vendor-finance/balances?addresses%5B%5D=0x245cc372C84B3645Bf0Ffe6538620B04a217988B&network=ethereum`
5. Verify v1 on Arbitrum: `http://localhost:5001/apps/vendor-finance/balances?addresses%5B%5D=0xD6587a974C7D3ecE23Fa53d5606da6B291311F6f&network=arbitrum`

### CODE REVIEW

**Comment by wpoulin** [comment]
Is there something wrong with fetching the balance onchain directly ? @ForrestChew

**Comment by ForrestChew** [comment]
> Is there something wrong with the current implementation fetching the balance onchain directly ? @ForrestChew We don't want to rely on subgraph or any external API for balances

Regarding the balances for both V2 ETH and V2 ARB, there have been no modifications, and they continue to be fetched from onchain as before. However, for fetching the actual pools, we must rely on an external API as there is no onchain method for tracking individual pools.

For both V1 ETH and V1 ARB balances, I'll try and find a way around using an external API. In regards to fetching the pools for V1, it is the same as V2.

File: src/apps/vendor-finance/common/vendor-finance.pool.contract-position-fetcher.ts [hunk 1]
```diff
@@ -149,12 +143,18 @@ export abstract class VendorFinancePoolContractPositionFetcher extends ContractP
...
 
     // --- Borrower logic ----
-    const [borrowPosition, mintRatioRaw] = await Promise.all([contract.debt(address), contract.mintRatio()]);
+    const data: VendorBorrowerGraphResponse = await Axios.post(this.entityUrl, {
+      query: borrowerInfosQuery(address),
+      type: `v1-${this.network.charAt(0).toUpperCase() + this.network.slice(1)}`,
+    });
+
+    const borrowerPosition = data.data.borrower?.positions.find(({ pool }) => pool.id === contractPosition.address);
+    if (!borrowerPosition) return [];
```
Comment (wpoulin) [inline src/apps/vendor-finance/common/vendor-finance.pool.contract-position-fetcher.ts:152]: Doesn't `contract.debt(address) ` return the borrow position ?

Comment (ForrestChew) [reply src/apps/vendor-finance/common/vendor-finance.pool.contract-position-fetcher.ts]: Sorry, got bogged down with something earlier. 

Yes you are right. I'm not entirely sure what prompted me to make the unfavorable switch in the first place. My mistake. I know you want to move away from external API calls whenever possible so I'll be more cognizant of that.

**Review by wpoulin — APPROVED** [review]

**Comment by wpoulin** [comment]
I'm wondering if you have Eslint running because I'm running it locally and there's a few linting errors ? @ForrestChew

**Comment by ForrestChew** [comment]
> 

So I get red linter errors for two things: 

1) `@Inject(APP_TOOLKIT)`
2) `@Inject(VendorFinanceContractFactory)`

I ignored these issues however because everything was still working as intended. Also, upon checking other projects within the studio, I noticed the same code and linter errors, leading me to believe it wasn't a real problem.

**Comment by wpoulin** [comment]
ohh okok, it's fine. I'll do the linting, merge and deploy asap thanks !

**Comment by ForrestChew** [comment]
> ohh okok, it's fine. I'll do the linting + smol change, merge and deploy asap thanks !

Appreciated. Thank you.

**Pull request merged**